packages/agreement-email-sender/src/services/agreementEmailSenderService.ts

```diff
@@ -96,34 +73,142 @@ async function getActivationMailFromAgreement(
     throw descriptorNotFound(agreement.eserviceId, agreement.descriptorId);
   }
 
-  return {
+  const mail = {
     subject: `Richiesta di fruizione ${agreement.id} attiva`,
     to: [producerEmail, consumerEmail],
     body: templateService.compileHtml(htmlTemplate, {
-      activationDate: formattedActivationDate,
+      activationDate,
       agreementId: agreement.id,
       eserviceName: eservice.name,
       eserviceVersion: descriptor.version,
       producerName: producer.name,
       consumerName: consumer.name,
     }),
   };
+
+  logger.info(`Sending email for agreement ${agreement.id} activation`);
+  await emailManager.send(
+    agreementEmailSender,
+    mail.to,
+    mail.subject,
+    mail.body
+  );
+  logger.info(`Email sent for agreement ${agreement.id} activation`);
 }
 
-// eslint-disable-next-line max-params
-export async function sendAgreementEmail(
-  agreement: AgreementV2,
+export async function sendAgreementSubmissionMail(
+  agreementV2: AgreementV2,
   readModelService: ReadModelService,
   emailManager: EmailManager,
   logger: Logger,
-  { agreementEmailSender } = agreementEmailSenderConfig()
+  { awsSesAgreementEmailSender } = agreementEmailSenderConfig()
 ): Promise<void> {
-  const { to, subject, body } = await getActivationMailFromAgreement(
+  const agreement = fromAgreementV2(agreementV2);
+  const templateService = buildHTMLTemplateService();
```

Comment:
is it required to create this service for each message?
(same for `sendAgreementActivationEmail`)